deep-learning/src/main/python/synapse/ml/dl/LitDeepTextModel.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in project root for information.
+
+import inspect
+
+import pytorch_lightning as pl
+import torch
+import torch.nn.functional as F
+import torch.optim as optim
+from pytorch_lightning.utilities import _module_available
+
+_TRANSFORMERS_AVAILABLE = _module_available("transformers")
+if _TRANSFORMERS_AVAILABLE:
+    import transformers
+
+    _TRANSFORMERS_EQUAL_4_15_0 = transformers.__version__ == "4.15.0"
+    if _TRANSFORMERS_EQUAL_4_15_0:
+        from transformers import AutoModelForSequenceClassification
+    else:
+        raise RuntimeError(
+            "transformers should be == 4.15.0, found: {}".format(
+                transformers.__version__
+            )
+        )
+else:
+    raise ModuleNotFoundError("module not found: transformers")
+
+
+class LitDeepTextModel(pl.LightningModule):
+    def __init__(
+        self,
+        checkpoint,
+        text_col,
+        label_col,
+        num_labels,
+        additional_layers_to_train,
+        optimizer_name,
+        loss_name,
+        train_from_scratch=True,  # train from scratch gives the best performance
+    ):
+        super(LitDeepTextModel, self).__init__()
+
+        self.checkpoint = checkpoint
+        self.text_col = text_col
+        self.label_col = label_col
+        self.num_labels = num_labels
+        self.additional_layers_to_train = additional_layers_to_train
+        self.optimizer_name = optimizer_name
+        self.loss_name = loss_name
+        self.train_from_scratch = train_from_scratch
+
+        self._check_params()
+
+        self.save_hyperparameters(
+            "checkpoint",
+            "text_col",
+            "label_col",
+            "num_labels",
+            "additional_layers_to_train",
+            "optimizer_name",
+            "loss_name",
+            "train_from_scratch",
+        )
```

Comment:
i think you dont need this list of args, it defaults to everything in constructor

Reply:
Emmm I think I need it for the current version of pytorch_lightning, because I found a bug in pytorch_lightning of saving all hyperparameters in both the current class and the class containing it. So we could have a try of their fix with a newer version, and if that works we can remove this save_hyperparameters then.

Reply:
Very weird, so long as you are aware I am fine with either!